ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -955,53 +1063,126 @@ namespace OrdinalAxis {
+                // final return value is based on ordinalIndex
             } else {
 
-                let ordinalLength = ordinalPositions.length,
-                    i,
-                    distance,
-                    ordinalIndex;
+                if (!extendedOrdinalPositions) {
+                    extendedOrdinalPositions =
+                        ordinal.getExtendedPositions &&
+                        ordinal.getExtendedPositions();
+                    ordinal.extendedOrdinalPositions = extendedOrdinalPositions;
+                }
+                if (!(extendedOrdinalPositions && extendedOrdinalPositions.length)) {
 
-                // first look for an exact match in the ordinalpositions array
-                i = ordinalLength;
-                while (i--) {
-                    if (ordinalPositions[i] === val) {
-                        ordinalIndex = i;
-                        break;
-                    }
+                    return val;
                 }
 
-                // if that failed, find the intermediate position between the
-                // two nearest values
-                i = ordinalLength - 1;
-                while (i--) {
-                    if (val > ordinalPositions[i] || i === 0) { // interpolate
-                        // something between 0 and 1
-                        distance = (val - ordinalPositions[i]) /
-                            (ordinalPositions[i + 1] - ordinalPositions[i]);
-                        ordinalIndex = i + distance;
-                        break;
+                let EOPlength = extendedOrdinalPositions.length,
+                    extendedOrdinalIndex;
+
+                if (!slope) {
+                    slope =
+                        (extendedOrdinalPositions[EOPlength - 1] -
+                            extendedOrdinalPositions[0]) /
+                        EOPlength;
+                }
+                // this value represents the value, where the first index of
+                // ordinalPositions lays in extendedOrdinalPositions.
+                // in other words, how much the OP is moved inside the EOP
+
+                const originalPositionsReference = Composition.findIndexOf(
+                    extendedOrdinalPositions,
+                    ordinalPositions[0]
+                );
+                // If the searched value is outside the visiblePlotArea,
+                // check if it is inside ExtendedOrdinalPositions.
+                if (
+                    val >= extendedOrdinalPositions[0] &&
+                    val <=
+                        extendedOrdinalPositions[
+                            extendedOrdinalPositions.length - 1
+                        ]
+                ) {
+                    let index = Composition.findIndexOf(
+                        extendedOrdinalPositions,
+                        val,
+                        true
+                    );
+
+                    if (extendedOrdinalPositions[index] === val) {
```

Comment:
As it was in the `bugfix/13238` branch, I think here we could apply the same.
Replace the `extendedOrdinalPositions` and `ordinalPositions` with a clever switch because the algorithm seems to be the same in both cases.  The array is the only thing that changes.